tests/functional/test_install_reqs.py

```diff
@@ -256,3 +256,14 @@ def test_constraints_only_causes_error(script, data):
         'install', '--no-index', '-f', data.find_links, '-c',
         script.scratch_path / 'c.txt', expect_error=True)
     assert 'installed requiresupper' not in result.stdout
+
+
+def test_constraints_local_editable_install_causes_error(script, data):
+    script.scratch_path.join("constraints.txt").write(
+        "Local==0.0.1"
+    )
+    to_install = data.packages.join("Local")
+    result = script.pip(
+        'install', '-c', script.scratch_path / 'constraints.txt', '-e',
+        to_install, expect_error=True)
+    assert 'Editable packages cannot be constrained' in result.stderr
```

Comment:
this looks good. please make a second variation without the -e. (Both PATH-TO-THING and -e PATH-TO-THING should error in the same basic fashion).
please use --no-index and -f data/packages to make the test fast. running py.test -k test_constraints_local_editable_install_causes_error should be a second or so -tops- that way.